lntest/itest/lnd_test.go

```diff
@@ -14106,6 +14106,97 @@ func testHoldInvoicePersistence(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 }
 
+// respondChannelAccept receives a *lnrpc.ChannelAcceptRequest from the
+// funding manager and responds with a predicate telling the funding manager
+// to accept or reject the open channel request.
+func respondChannelAccept(predStream lnrpc.Lightning_ChannelAcceptorClient,
+	response *lnrpc.ChannelAcceptResponse, t *harnessTest) {
+	go func() {
+		// Opening the channel should send a *lnrpc.ChannelAcceptRequest to the stream.
+		req, err := predStream.Recv()
+		if err != nil {
+			t.Fatalf("unable to receive open channel request: %v", err)
+		}
+
+		// If a response was passed, we send it to the stream.
+		if response != nil {
+			response.PendingChanId = req.PendingChanId
+			if err := predStream.Send(response); err != nil {
+				t.Fatalf("unable to accept open channel request: %v", err)
+			}
+		}
+	}()
+}
+
+func testPredicateChannelAccept(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	ctxt, cancel := context.WithCancel(ctxb)
+	defer cancel()
+
+	// Open up a bidirectional stream so that Bob can respond to incoming
+	// open channel requests.
+	bobPredStream, err := net.Bob.ChannelAcceptor(ctxt)
+	if err != nil {
+		t.Fatalf("unable to create channel acceptor stream for bob: %v", err)
+	}
+
+	// Accept the channel request.
+	acceptReponse := &lnrpc.ChannelAcceptResponse{
+		Accept: true,
+	}
+
+	respondChannelAccept(bobPredStream, acceptReponse, t)
+
+	// Here we are going to test an accepted channel open.
+	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+	chanPoint := openChannelAndAssert(
+		ctxt, t, net, net.Alice, net.Bob,
+		lntest.OpenChannelParams{
+			Amt: 100000,
+		},
+	)
+
+	// Here we close the accepted channel.
+	ctxt, _ = context.WithTimeout(ctxb, channelCloseTimeout)
+	closeChannelAndAssert(ctxt, t, net, net.Alice, chanPoint, false)
+
+	// Reject the channel request.
+	rejectReponse := &lnrpc.ChannelAcceptResponse{
+		Accept: false,
+	}
+
+	respondChannelAccept(bobPredStream, rejectReponse, t)
+
+	// Here we are going to test a rejected channel open.
+	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+	_, err = net.OpenChannel(
+		ctxt, net.Alice, net.Bob,
+		lntest.OpenChannelParams{
+			Amt: 100000,
+		},
+	)
+	if err == nil || grpc.Code(err) != 202 {
+		t.Fatalf("failed to reject channel: %v", err)
+	}
+
+	// Timeout the channel request.
```

Comment:
`DefaultRPCTimeout` = 15 seconds

Reply:
That's a bit long for a integration test, since it will just sit here?

Reply:
I think the only way to configure this and have the same code coverage would be to specify the timeout when calling the `ChannelAcceptor` rpc method.  I do think it gives the user more control over this new functionality, what do you think?

Reply:
Yeah, would that work, or would it be hard because of the bidirectional stream? 

I think either way that it should be configurable, as imagine you want to run some fancy algorithm to determine whether to accept the channel, that is super heavy :P

Reply:
Should be possible w/ the bidi stream, I agree it should be configurable.  Will report back

Reply:
Since there can only be one input and one output, we can't really specify the timeout unless it's part of the `OpenChannelResponse`, at which point the timeout would apply to the *next* `OpenChannelRequest` message.  I think this approach is confusing.  The timeout could alternatively be specified as a separate rpc or as a command line option when starting lnd.  The downside of the command line option is that it's not mutable unless the user restarts their lnd instance.